src/main/scala/rocketchip/Configs.scala

```diff
@@ -115,7 +115,10 @@ class BasePlatformConfig extends Config (
           idBits = Dump("MEM_ID_BITS", site(MIFTagBits)))
       }
       case BuildCoreplex => (p: Parameters) => Module(new DefaultCoreplex(p))
-      case NExtInterrupts => 2
+      case NExtTopInterrupts => 2
+      case NExtPeripheryInterrupts => 0
```

Comment:
Look at how nClientPorts is used in the DeviceBlock class.

Basically, I'm suggesting that you add an nInterrupts method to the DeviceBlock class. Then you change this line to be

    case NExtPeripheryInterrupts => site(ExtraDevices).nInterrupts